lib/Search/PersonSearchProvider.php

```diff
@@ -0,0 +1,147 @@
+ *
+ * @author Li Xiangbin <dassio@icloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\FaceRecognition\Search;
+
+use OCP\Search\IProvider;
+use OCP\IL10N;
+use OCP\IUser;
+use OCP\Search\ISearchQuery;
+use OCP\Search\SearchResult;
+use OCP\Search\SearchResultEntry;
+use OCP\Files\IRootFolder;
+use OCP\IURLGenerator;
+use OCP\Files\IMimeTypeDetector;
+
+use OCA\FaceRecognition\Db\Person;
+use OCA\FaceRecognition\Db\PersonMapper;
+use OCA\FaceRecognition\Db\ImageMapper;
+use OCA\FaceRecognition\Service\SettingsService;
+use OCA\FaceRecognition\Model\IModel;
+
+/**
+ * Provide search results from the 'facerecognition' app
+ */
+class PersonSearchProvider implements IProvider {
+
+	/** @var PersonMapper personMapper */
+	private $personMapper;
+
+	/** @var ImageMapper imageMapper */
+	private $imageMapper;
+
+	/** @var SettingsService Settings service */
+	private $settingsService;
+
+	/** @var IL10N */
+	private $l10n;
+
+	/** @var IMimeTypeDetector */
+	private $mimeTypeDetector;
+
+	/** @var IURLGenerator */
+	private $urlGenerator;
+
+	/** @var IRootFolder */
+	private $rootFolder;
+
+	/** @var IModel*/
+	private $modelId;
+
+	public function __construct(PersonMapper      $personMapper,
+	                            ImageMapper       $imageMapper,
+	                            SettingsService   $settingsService,
+	                            IL10N             $l10n,
+	                            ImimeTypeDetector $mimeTypeDetector,
+	                            IURLGenerator     $urlGenerator,
+	                            IRootFolder       $rootFolder) {
+		$this->personMapper     = $personMapper;
+		$this->imageMapper = $imageMapper;
+		$this->settingsService = $settingsService;
+		$this->l10n = $l10n;
+		$this->urlGenerator = $urlGenerator;
+		$this->rootFolder = $rootFolder;
+		$this->mimeTypeDetector = $mimeTypeDetector;
+		$this->modelId =$this->settingsService->getCurrentFaceModel();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getId(): string {
+		return 'facerecognition';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getName(): string {
+		return $this->l10n->t('Face Recognition');
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getOrder(string $route, array $routeParameters): int {
+		return 10;
```

Comment:
Restore optional priority.
It should be a high priority if you are within the application, and low anywhere else. :smile: